art/attacks/attack.py

```diff
@@ -154,7 +155,8 @@ def __init__(self):
         """
         Initializes black-box data poisoning attack
         """
-        super().__init__(None)
+        empty_black_box = BlackBoxClassifier(None, None, 0)
+        super().__init__(empty_black_box)
```

Comment:
I think this change is because the `Attack` base class expects a classifier which is not None. `Attack` did not expect the `PoisoningAttackBlackBox` API so far which does not require a classifier. I would like to propose to update the check in `Attack` to accept `classifier=None` instead of using `empty_black_box`: https://github.com/IBM/adversarial-robustness-toolbox/blob/e6d0b53a16bca37712a78ad916018c67a5951abe/art/attacks/attack.py#L42

Reply:
Fixed in commit